packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -587,7 +597,9 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
 }
 
 export class RemotedFluidDataStoreContext extends FluidDataStoreContext {
-    private details: ISnapshotDetails | undefined;
+    private readonly intialSnapshotDetailsP =
+        new LazyPromise<ISnapshotDetails>(async () => this.loadInitialSnapshotDetails());
```

Comment:
>loadInitialSnapshotDetails [](start = 59, length = 26)

i would avoid calling member function here, as they don't really makes sense as members as they should never be called elsewhere . if it's small just inline it, or move out of the class to a non-exported function

Reply:
These methods are not very big so I inlined them here.